app/javascript/src/controller_branches.js

```diff
@@ -171,6 +177,26 @@ BranchesController.createBranch = function($node) {
   return branch;
 }
 
+BranchesController.addBranchCombinator = function(args) {
+  var branch = args[0];
+  if( !branch.index == 0 ) {
+    branch.$node.before("<p class=\"branch-or\">or</p>");
+  }
+}
+
+BranchesController.removeBranchCombinator = function(node) {
+    var $or = $(node).prev('.branch-or').first();
+    if($or) {
+      $or.remove()
+    }
+}
+
+function addBranchEventListeners(view) {
+  view.$document.on('branchRemoved', function(event, node){
```

Comment:
A quick `grep -r "\$document\.on" app/javascript/src` shows these results:
```
view.$document.on("AddComponentMenuSelection", AddComponent.MenuSelection.bind(view) );
this.$document.on("SaveRequired", () => dataController.saveRequired(true) );
view.$document.on("QuestionMenuSelectionRemove", function(event, question) {
view.$document.on("QuestionMenuSelectionRequired", function(event, question) {
view.$document.on("ContentMenuSelectionRemove", function(event, component) {
```

I'm surprised because I expected them to be all lowercase (which is why I checked in the first place). Either way, although a small point it's probably worth sticking with the upper camel case/pascal case version for code consistency.

I'm also guessing that it's the Branch.remove method that is triggering the event so, not that I'd say there is any hard and fast event naming convention but, it would seem, we can let one evolve. Looking at others, maybe it's evolving to be ThingMethod or ThingActionMethod ??

'BranchRemove' seems closest to establishing/keeping a convention of some sort.